examples/graph/gnn_citations.py

```diff
@@ -0,0 +1,705 @@
+    return model
+
+
+baseline_model = create_baseline_model(hidden_units, num_classes, dropout_rate)
+baseline_model.summary()
+
+"""
+### Train the baseline classifier
+"""
+
+history = run_experiment(baseline_model, x_train, y_train)
+
+"""
+Let's plot the learning curves.
+"""
+
+display_learning_curves(history)
+
+"""
+Now we evaluate the baseline model on the test data split.
+"""
+
+_, test_accuracy = baseline_model.evaluate(x=x_test, y=y_test, verbose=0)
+print(f"Test accuracy: {round(test_accuracy * 100, 2)}%")
+
+"""
+### Examine the baseline model predictions
+
+Let's create new data instances by randomly generating binary word vectors with respect to
+the word presence probabilities.
+"""
+
+
+def generate_random_instances(num_instances):
+    token_probability = x_train.mean(axis=0)
+    instances = []
+    for _ in range(num_instances):
+        probabilities = np.random.uniform(size=len(token_probability))
+        instance = (probabilities <= token_probability).astype(int)
+        instances.append(instance)
+
+    instances = np.array(instances)
+    return instances
+
+
+def display_class_probabilities(probabilities):
+    for instance_idx, probs in enumerate(probabilities):
+        print(f"Instance {instance_idx + 1}:")
+        for class_idx, prob in enumerate(probs):
+            print(f"- {class_values[class_idx]}: {round(prob * 100, 2)}%")
+        print("")
+
+
+"""
+Now we show the baseline model predictions given these randomly generated instance.
+"""
+
+new_instances = generate_random_instances(num_classes)
+logits = baseline_model.predict(new_instances)
+probabilities = keras.activations.softmax(tf.convert_to_tensor(logits)).numpy()
+display_class_probabilities(probabilities)
+
+"""
+## Build a Graph Neural Network Model
+
+### Prepare the data for the graph model
+
+Preparing and loading the graphs data into the model for training is the most challenging
+part in GNN models, which is addressed in different ways by the specialised libraries.
+In this example, we show a simple approach for preparing and using graph data that is suitable
+if your dataset consists of a single graph that fits entirely in memory.
+
+We graph data is represented by the `graph_info` tuple, which consists of the following
+three elements:
+
+1. `node_features`: This is a `[num_nodes, num_features]` NumPy array that includes the
+node features. In this dataset, the nodes are the papers, and the `node_features` are the
+word-presence binary vectors of each paper.
+2. `edges`:  This is `[num_edges, num_edges]` NumPy array representing a sparse
+[adjacency matrix](https://en.wikipedia.org/wiki/Adjacency_matrix#:~:text=In%20graph%20theory%20and%20computer,with%20zeros%20on%20its%20diagonal.)
+of the links between the nodes. In this example, the links are the citations between the papers.
+3. `edge_weights` (optional): This is a `[num_edges]` NumPy array that includes the edge weights.
```

Comment:
I got it its not used in this example but what edge weights would be otherwise?

Reply:
OK updated